charts/revere/templates/workloadIdentity.yaml

```diff
@@ -0,0 +1,27 @@
+{{- if .Values.secrets.gcpServiceAccount.workloadIdentity.enabled }}
+---
+# Will reference SA if it already exists, creates otherwise;
+# we don't have control over that behavior.
+# https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMServiceAccount
+metadata:
+  name: {{ .Values.secrets.gcpServiceAccount.workloadIdentity.accountName }}
+  labels:
+    {{- include "revere.labels" . | nindent 4 }}
+---
+apiVersion: iam.cnrm.cloud.google.com/v1beta1
+kind: IAMPolicyMember
+metadata:
+  name: {{ .Values.name }}-iampolicy-member
+  labels:
+    {{- include "revere.labels" . | nindent 4 }}
+spec:
+  member: serviceAccount:{{ .Values.secrets.gcpServiceAccount.workloadIdentity.projectId }}.svc.id.goog[{{ .Release.Namespace }}/{{ .Values.name }}-sa]
```

Comment:
How does the namespace get used here? If this is necessary so be it, but it's good to be careful when using `.Release.Namespace` is it can unintentional bake in assumptions about where an application is deployed.

Reply:
I realize it shows up in some other helm charts too 😄 That is a small tech debt issue to address.

Reply:
It is unfortunately very necessary, [this](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#identity_sameness_across_clusters) helps explain why and [step 7 here](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to) demos it being necessary whether you use gcloud or a manifest to do it.